pychiquito/dsl.py

```diff
@@ -22,6 +22,7 @@ def __init__(self: Circuit):
         self.ast = ASTCircuit()
         self.witness = TraceWitness()
         self.rust_ast_id = 0
+        self.num_step_instances = 0
```

Comment:
@qwang98 remember that all the witness generation fields have to be reset after a witness is generated because we can generate more than a witness for the circuit

Reply:
This isn't an issue, because `TraceWitness()` constructor creates an empty default witness before calling `trace`:

```
def gen_witness(self: Circuit, args: Any) -> TraceWitness:
        self.mode = CircuitMode.Trace
        self.witness = TraceWitness()
        # ...
```

Reply:
Yes, but it does not set num_step_instances to 0.